test/autocast_fp8.cpp

```diff
@@ -0,0 +1,172 @@
+        auto y_fp8       = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::fp8e4m3fnuz_type}}), y_fp32);
+        auto x_fp8       = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::fp8e4m3fnuz_type}}), x_fp32);
+        auto sum_fp8     = m2.add_instruction(migraphx::make_op("add"), x_fp8, y_fp8);
+        auto diff_fp8    = m2.add_instruction(migraphx::make_op("sub"), x_fp8, y_fp8);
+        auto sum_fp32    = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::float_type}}), sum_fp8);
+        auto diff_fp32   = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::float_type}}), diff_fp8);
+        m2.add_return({sum_fp32, diff_fp32});
+    }
+    EXPECT(m1 == m2);
+}
+
+// multiple inputs (of different types) to return
+TEST_CASE(autocast_fp8_4)
+{
+    migraphx::module m1;
+    {
+        auto x1      = m1.add_parameter("x1", {migraphx::shape::fp8e4m3fnuz_type, {1}});
+        auto y1      = m1.add_parameter("y1", {migraphx::shape::fp8e4m3fnuz_type, {1}});
+        auto x2      = m1.add_parameter("x2", {migraphx::shape::float_type, {1}});
+        auto y2      = m1.add_parameter("y2", {migraphx::shape::float_type, {1}});
+        auto sum1    = m1.add_instruction(migraphx::make_op("add"), x1, y1);
+        auto sum2    = m1.add_instruction(migraphx::make_op("add"), x2, y2);
+        m1.add_return({sum1, sum2});
+    }
+    run_pass(m1);
+
+    migraphx::module m2;
+    {
+        auto x2          = m2.add_parameter("x2", {migraphx::shape::float_type, {1}});
+        auto y2          = m2.add_parameter("y2", {migraphx::shape::float_type, {1}});
+        auto y1          = m2.add_parameter("y1", {migraphx::shape::float_type, {1}});
+        auto x1          = m2.add_parameter("x1", {migraphx::shape::float_type, {1}});
+        auto y1_fp8      = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::fp8e4m3fnuz_type}}), y1);
+        auto x1_fp8      = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::fp8e4m3fnuz_type}}), x1);
+        auto sum1        = m2.add_instruction(migraphx::make_op("add"), x1_fp8, y1_fp8);
+        auto sum2        = m2.add_instruction(migraphx::make_op("add"), x2, y2);
+        auto result_sum1 = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::float_type}}), sum1);
+        m2.add_return({result_sum1, sum2});
+    }
+    EXPECT(m1 == m2);
+}
+
+// autocast pass does not do any changes
+TEST_CASE(autocast_fp8_5)
+{
+    migraphx::module m1;
+    {
+        auto x   = m1.add_parameter("x", {migraphx::shape::float_type, {1}});
+        auto y   = m1.add_parameter("y", {migraphx::shape::float_type, {1}});
+        auto sum = m1.add_instruction(migraphx::make_op("add"), x, y);
+        m1.add_return({sum});
+    }
+    run_pass(m1);
+
+    migraphx::module m2;
```

Comment:
you can remove this m2 and 
do `run_pass(m1) == m1`

Reply:
Not sure if I am missing something but I tried this and get the error ` error: invalid operands to binary expression ('void' and 'migraphx::module')
    run_pass(m1) == m1;`.

Reply:
oh i see. you would need to copy module m1 first. 

so `module m2 = m1` and then do `run_pass(m1)` and do `m1 == m2`.